src/pycram/world_reasoning.py

```diff
@@ -62,6 +63,56 @@ def contact(
             return objects_are_in_contact
 
 
+def robot_will_be_in_collision_at_pose(robot: Object, pose: Pose,
+                                       ignore_collision_with: Optional[List[Object]] = None) -> bool:
+    """
+    Check if the robot collides with any object in the world at the given pose.
+
+    :param robot: The robot object
+    :param pose: The pose to check for collision
+    :param ignore_collision_with: A list of objects to ignore collision with
+    :return: True if the robot collides with any object, False otherwise
+    """
+    with UseProspectionWorld():
+        prospection_robot = World.current_world.get_prospection_object_for_object(robot)
+        prospection_robot.set_pose(pose)
+        floor = prospection_robot.world.get_object_by_name("floor")
+        ignore_collision_with = [] if ignore_collision_with is None else ignore_collision_with
+        ignore = [o.name for o in ignore_collision_with]
+        for obj in prospection_robot.world.objects:
+            if obj.name in ([prospection_robot.name, floor.name] + ignore):
+                continue
+            in_contact, contact_links = contact(prospection_robot, obj, return_links=True)
+            if in_contact and not is_a_picked_object(prospection_robot, obj, [links[0] for links in contact_links]):
+                logdebug(f"Robot is in contact with {obj.name} in prospection: {obj.world.is_prospection_world}"
+                         f"at position {pose.position_as_list()} and z_angle {pose.z_angle}")
+                return True
+            logdebug(f"Robot is not in contact with {obj.name} in prospection: {obj.world.is_prospection_world}"
+                     f"at position {pose.position_as_list()} and z_angle {pose.z_angle}")
+    return False
+
+
+def is_a_picked_object(robot: Object, obj: Object, robot_contact_links: List[Link]) -> bool:
```

Comment:
I think is_held_object is  a better name

Reply:
Done